### PR TITLE
Mushking Royale Stats and Medals

### PR DESCRIPTION
**Pull request by Zintixx**
Implements MKR stats and medals.
Do note that you would need to edit /Xml/table/na/maplesurvivalopenperiod.xml to a date in the future to test these out. The vanilla XML will not work and you will not be able to open the MKR UI.

We're not parsing survivallevelreward.xml because this is the old MKR system (used in KMS2 and during GMS2 betas).

### CODE REVIEW

**Review by darkvergus — APPROVED** [review]

File: MapleServer2/Database/SQL/Database.sql [hunk 1]
```diff
@@ -723,6 +728,47 @@ CREATE TABLE `ugc_market_sales`
...
+  COLLATE = utf8mb4_0900_ai_ci;
+
+--
+-- Table structure for table `medals`
+--
+
+DROP TABLE IF EXISTS `medals`;
+CREATE TABLE `medals`
+(
+    `uid`                              bigint           NOT NULL AUTO_INCREMENT,
```
Comment (JamieDavidson) [inline MapleServer2/Database/SQL/Database.sql:756]: Can we name this `id` to match other tables please?

File: MapleServer2/Database/Classes/DatabaseMedal.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        return QueryFactory.Query(TableName).Where("id", uid).Delete() == 1;
+    }
+
+    private static Medal ReadMedal(dynamic data)
+    {
+        return new Medal()
+        {
+            Uid = data.uid,
+            Id = data.effect_id,
```
Comment (JamieDavidson) [inline MapleServer2/Database/Classes/DatabaseMedal.cs:53]: If Id is an effect Id the property should be renamed to EffectId IMO

**Review by JamieDavidson — CHANGES_REQUESTED** [review]

File: MapleServer2/PacketHandlers/Game/MushkingRoyaleSystemHandler.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+    private static void UnequipMedal(GameSession session, MedalSlot slot)
+    {
+        Medal oldMedal = session.Player.Account.EquippedMedals[slot];
+        if (oldMedal is not null)
+        {
+            oldMedal.IsEquipped = false;
+            DatabaseManager.MushkingRoyaleMedals.Update(oldMedal);
+        }
+    }
```
Comment (AngeloTadeucci) [inline MapleServer2/PacketHandlers/Game/MushkingRoyaleSystemHandler.cs:81]: Move this method to Account

**Review by AngeloTadeucci — CHANGES_REQUESTED** [review]

File: MapleServer2/Types/MushkingRoyale/MushkingRoyaleStats.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                newExp = 0;
+                break;
+            }
+        }
+
+        Exp = newExp;
+        session.Send(MushkingRoyaleSystemPacket.LoadStats(this, newExp));
+    }
+
+    public bool LevelUp()
```
Comment (AlanMorel) [inline MapleServer2/Types/MushkingRoyale/MushkingRoyaleStats.cs:56]: Can this be private if this will never be needed outside of this class?

**Review by AlanMorel — APPROVED** [review]

File: MapleServer2/Database/SQL/Database.sql [hunk 5]
```diff
@@ -723,6 +728,47 @@ CREATE TABLE `ugc_market_sales`
...
   COLLATE = utf8mb4_0900_ai_ci;
 
+--
+-- Table structure for table `mushking_royale_stats`
+--
+
+DROP TABLE IF EXISTS `mushking_royale_stats`;
+CREATE TABLE `mushking_royale_stats`
+(
+    `id`                               bigint       NOT NULL AUTO_INCREMENT,
```
Comment (AngeloTadeucci) [inline MapleServer2/Database/SQL/Database.sql:738]: missed rename here

Comment (Zintixx) [reply MapleServer2/Database/SQL/Database.sql:738]: that's stats. id should be id in stats. i made the rename changes on medals

**Review by Sparkymod — APPROVED** [review]

File: MapleServer2/Database/SQL/Database.sql (hunk 5 above)
Comment (Sparkymod) [reply MapleServer2/Database/SQL/Database.sql:738]: id should be ok.

Comment (AngeloTadeucci) [reply MapleServer2/Database/SQL/Database.sql:738]: oh ok

**Pull request merged**